src/query/stream_to_xasl.c

```diff
@@ -2413,6 +2419,40 @@ stx_build_cte_xasl_id (THREAD_ENTRY * thread_p, char *ptr, XASL_ID * cte_xasl_id
   return ptr;
 }
 
+static char *
+stx_build_sq_cache (THREAD_ENTRY * thread_p, char *ptr, SQ_CACHE ** sq_cache_p)
+{
+  XASL_UNPACK_INFO *xasl_unpack_info = get_xasl_unpack_info_ptr (thread_p);
+  SQ_CACHE *new_sq_cache = NULL;
+  int n_elements, offset;
+
+  *sq_cache_p = NULL;
+  ptr = or_unpack_int (ptr, &n_elements);
+  ptr = or_unpack_int (ptr, &offset);
+
+  if (n_elements > 0)
+    {
+      new_sq_cache = (SQ_CACHE *) stx_alloc_struct (thread_p, sizeof (SQ_CACHE));
+      if (new_sq_cache == NULL)
+	{
+	  stx_set_xasl_errcode (thread_p, ER_OUT_OF_VIRTUAL_MEMORY);
+	  return NULL;
+	}
+      new_sq_cache->sq_key_struct =
```

Comment:
Exception handling is required.

Reply:
I've re-designed that function. Please confirm.